mitiq/ddd/rules/tests/test_rules.py

```diff
@@ -14,3 +14,89 @@
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
 """Unit tests for DDD rules."""
+from mitiq.ddd.rules.rules import construct_rule, xx, xyxy, yy
+import pytest
+from cirq import X, Y, Z
+
+phi = (1 + 5**0.5) / 2.0
+
+
+@pytest.mark.parametrize(
+    "slack_length",
+    [int(round((phi**n - (1 - phi) ** n) / 5**0.5)) for n in range(2, 10)],
+)
+def test_rules(slack_length):
+    @pytest.mark.parametrize(
+        "rule",
+        [
+            xx(slack_length),
+            xyxy(slack_length),
+            yy(slack_length),
+        ],
+    )
+    def test_rule(rule):
+        assert len(rule) == slack_length
```

Comment:
```suggestion
    assert len(rule) == slack_length
```

In other Mitiq tests we usually don't define test sub functions when using multiple `pytest.mark.parametrize`, we just stack multiple `pytest.mark.parametrize` decorators and define a unique test function with multiple arguments. But actually I am not sure what is the best practice.

Reply:
I included the sub functions here because the test parametrization(s) require `slack_length`. I don't remember best practice either at the moment.

Reply:
Apparently sub function tests obfuscated some errors, so I de-nested.